src/lib/message/L2ToL1Message.ts

```diff
@@ -65,104 +66,62 @@ export enum L2ToL1MessageStatus {
 export type L2ToL1MessageReaderOrWriter<T extends SignerOrProvider> =
   T extends Provider ? L2ToL1MessageReader : L2ToL1MessageWriter
 
-export type L2ToL1TransactionEvent = {
-  caller: string
-  destination: string
-  hash: BigNumber
-  position: BigNumber
-  indexInBatch?: BigNumber
-  arbBlockNum: BigNumber
-  ethBlockNum: BigNumber
-  timestamp: BigNumber
-  callvalue: BigNumber
-  data: string
-}
-
 /**
  * Base functionality for L2->L1 messages
  */
 export class L2ToL1Message {
+  protected isClassic(
+    e: L2ToL1TransactionEvent
+  ): e is ClassicL2ToL1TransactionEvent['args'] {
+    if ((e as ClassicL2ToL1TransactionEvent['args']).indexInBatch) return true
+    else return false
+  }
+
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event: L2ToL1TransactionEvent,
-    outboxAddress: string
+    event: NitroL2ToL1TransactionEvent['args']
   ): L2ToL1MessageReaderOrWriter<T>
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event: L2ToL1TransactionEvent,
+    event: ClassicL2ToL1TransactionEvent['args'],
     outboxAddress: string
+  ): L2ToL1MessageReaderOrWriter<T>
+  static fromEvent<T extends SignerOrProvider>(
+    l1SignerOrProvider: T,
+    event: L2ToL1TransactionEvent,
+    outboxAddress?: string
   ): L2ToL1MessageReader | L2ToL1MessageWriter {
     return SignerProviderUtils.isSigner(l1SignerOrProvider)
       ? new L2ToL1MessageWriter(l1SignerOrProvider, event, outboxAddress)
       : new L2ToL1MessageReader(l1SignerOrProvider, event, outboxAddress)
   }
 
-  public static async getL2ToL1MessageLogs(
+  public static async getEventLogs(
     l2Provider: Provider,
     filter: { fromBlock: BlockTag; toBlock: BlockTag },
-    batchNumber?: BigNumber,
+    positionOrBatchNumber?: BigNumber,
```

Comment:
I think its better if we split this into 2 separate parameters

Reply:
would be cool if you could add some typedoc here explaining when each parameter is expected

Reply:
Keeping it as one parameter maintains the same API. I'll add some documentation about why they're named this way.